bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/FlatRowAdapter.java

```diff
@@ -60,20 +78,42 @@ public Result adaptResponsePresortedCells(FlatRow flatRow) {
     if (flatRow == null) {
       return Result.EMPTY_RESULT;
     }
-
-    return convert(flatRow, new ArrayList<org.apache.hadoop.hbase.Cell>());
+    List<Cell> hbaseCells = addCells(flatRow, false);
+    return Result.create(hbaseCells.toArray(new Cell[hbaseCells.size()]));
   }
 
-  private Result convert(FlatRow flatRow, Collection<org.apache.hadoop.hbase.Cell> hbaseCells) {
+  /**
+   * Converts all of the {@link FlatRow#getCells()} into HBase {@link RowCell}s.
+   * <p>
+   * All @link FlatRow.Cell}s with labels are discarded, since those are used only for
+   * {@link WhileMatchFilter}
+   * <p>
+   * Some additional deduping and sorting is also required. Cloud Bigtable returns values sorted by
+   * family in lexicographically ascending order, an arbitrary ordering of qualifiers (by internal
+   * id rather than lexicographically) and finally by timestamp descending. Each cell can appear
+   * more than once, if there are {@link Interleave}s in the {@link ReadRowsRequest#getFilter()},
+   * but the duplicates will appear one after the other. An {@link Interleave}s can originate from a
+   * {@link FilterList} with a {@link FilterList.Operator#MUST_PASS_ONE} operator or a
+   * {@link WhileMatchFilter}.
+   * <p>
+   * @param flatRow
+   * @param sort
+   * @return
+   */
+  private List<Cell> addCells(FlatRow flatRow, boolean sort) {
```

Comment:
nit: convertCells()? addCells() sounds like adding cells to an input parameter (like the one we had with TreeSet).